exporter/prometheusremotewriteexporter/factory.go

```diff
@@ -45,7 +45,7 @@ func createMetricsExporter(_ context.Context, params component.ExporterCreatePar
 		return nil, errors.New("invalid configuration")
 	}
 
-	client, err := prwCfg.HTTPClientSettings.ToClient()
+	client, err := prwCfg.HTTPClientSettings.ToClient(map[config.ComponentID]component.Extension{})
```

Comment:
This is one of the reasons I think the refactorings to callers should be done as part of this PR. This call here for instance would have to be made during the start, and not in the factory. This might reveal other potential problems that you'd need to fix in the extension mechanism (client auth code).

Reply:
Sure will start the refactoring.